### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by sovcik**
Trivial declaration file for typescript so VS Code won't yell while editing webpack config file...

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for creating a TypeScript definition. 🙌 Can you follow [this styleguide](https://github.com/sindresorhus/typescript-definition-style-guide)?

Some examples:
- https://github.com/sindresorhus/semver-regex/commit/039944bb5b3876913171b895488874b5c200e458
- https://github.com/sindresorhus/global-dirs/commit/0aae4d64887f77af94c4fb7ca12aa9f93c0a8291
- https://github.com/sindresorhus/md5-hex/commit/bbbfeff840f26973fbffc16d4863ae6d120079f3

**Comment by sovcik** [comment]
Tried my best :-)

**Comment by sovcik** [comment]
Ok, not the best. Adding few things - like tds test...

File: index.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const ansiEscapes = require('ansi-escapes');
 const {supportsHyperlink} = require('supports-hyperlinks');
 const getRuleDocs = require('eslint-rule-docs');
 
-module.exports = (results, data) => {
+function format(results, data) {
```
Comment (sindresorhus) [inline index.js]: Don't do unrelated changes.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import {CLIEngine} from 'eslint';
+
+/**
+Format the lint results.
+
+@param results - Contains the lint results for individual files.
+@param data - Extended information related to the analysis results.
+@returns The formatted output.
+*/
+declare function format(results: CLIEngine.LintResult[], data?: {}): string;
```
Comment (sindresorhus) [inline index.d.ts]: `data` needs a more specific type.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import {CLIEngine} from 'eslint';
+
+/**
+Format the lint results.
+
+@param results - Lint result for the individual files.
+@param data - Extended information related to the analysis results.
```
Comment (sindresorhus) [inline index.d.ts]: The description could be more specific about what kind of extended information it expects.

**Comment by sovcik** [comment]
Well, the best would be to update @types/eslint first and then make format of type Formatter. I've created PR at DefinitelyTYped for that - will modify it here once is approved/merged there.

File: index.js (hunk 1 above)
Comment (sovcik) [reply index.js]: Reverted

File: index.d.ts (hunk 2 above)
Comment (sovcik) [reply index.d.ts]: Fixed in new @types/eslint

File: index.d.ts (hunk 3 above)
Comment (sovcik) [reply index.d.ts]: Fixed by new @types/eslint package

**Comment by sovcik** [comment]
I hope it's better now.

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+@returns The formatted output.
+*/
```
Comment (sindresorhus) [inline index.d.ts]: I think the doc comment should be added to the type in DefinitelyTyped. That way every consumer benefits from it and it can be properly kept in sync with ESLint changes.

Comment (sovcik) [reply index.d.ts]: Good idea. Please, go ahead and create PR.

Comment (sindresorhus) [reply index.d.ts:9]: I'm sorry, but I don't have time to do that. You are the one that wants this to be merged.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/eslint-formatter-pretty/issues/52
